Fix some regressions

hmenager opened the pull request.

hmenager committed:
quickfix to move record.py in the right folder

mr-c commented:
My testing notes:
Must use a Python 2.7 virtualenv to launch `tox -e cwl_green`

mr-c committed:
fix schemadef_req_tool_param & schemadef_req_wf_param

mr-c commented:
Down to just 8 regressions, also the tests take too long to run on Travis? Maybe if they pass they are faster :-)

mr-c commented:
Remaining tests:

> CwlConformanceTestCase.test_conformance_v1_0_directory_output
```
E           CompareFail: expected: {
E               "class": "Directory", 
E               "listing": [
E                   {
E                       "checksum": "sha1$dd0a4c4c49ba43004d6611771972b6cf969c1c01", 
E                       "class": "File", 
E                       "location": "goodbye.txt", 
E                       "size": 24
E                   }, 
E                   {
E                       "checksum": "sha1$47a013e660d408619d894b20806b1d5086aab03b", 
E                       "class": "File", 
E                       "location": "hello.txt", 
E                       "size": 13
E                   }
E               ]
E           }
E           got: {
E               "basename": "dir3.cwl on data 1", 
E               "class": "Directory", 
E               "listing": []
E           }
E           caused by: {
E               "checksum": "sha1$dd0a4c4c49ba43004d6611771972b6cf969c1c01", 
E               "class": "File", 
E               "location": "goodbye.txt", 
E               "size": 24
E           } not found
```

> CwlConformanceTestCase.test_conformance_v1_0_exprtool_directory_literal

(same empty Directory listing issue)

> CwlConformanceTestCase.test_conformance_v1_0_initial_workdir_empty_writable

(same empty Directory listing issue)

> CwlConformanceTestCase.test_conformance_v1_0_initial_workdir_empty_writable_docker

(same empty Directory listing issue)

> CwlConformanceTestCase.test_conformance_v1_0_initial_workdir_output
```
E           CompareFail: expected: {
E               "checksum": "sha1$aeb3d11bdf536511649129f4077d5cda6a324118", 
E               "class": "File", 
E               "location": "ref.fasta", 
E               "secondaryFiles": [
E                   {
E                       "checksum": "sha1$da39a3ee5e6b4b0d3255bfef95601890afd80709", 
E                       "class": "File", 
E                       "location": "ref.fasta.fai", 
E                       "size": 0
E                   }
E               ], 
E               "size": 12010
E           }
E           got: {
E               "basename": "ref.fasta", 
E               "checksum": "sha1$aeb3d11bdf536511649129f4077d5cda6a324118", 
E               "class": "File", 
E               "location": "ref.fasta", 
E               "nameext": ".fasta", 
E               "nameroot": "ref", 
E               "size": 12010
E           }
E           caused by: field 'secondaryFiles' failed comparison: expected: [
E               {
E                   "checksum": "sha1$da39a3ee5e6b4b0d3255bfef95601890afd80709", 
E                   "class": "File", 
E                   "location": "ref.fasta.fai", 
E                   "size": 0
E               }
E           ]
E           got: null
```

> CwlConformanceTestCase.test_conformance_v1_0_initial_workdir_secondary_files_expr
```
>               assert state in ok_states, "Final state - %s - not okay." % state
E               AssertionError: Final state - error - not okay.
```
```
WorkflowException: Missing required secondary file 'input.txt.idx1' from file object: {
    "nameroot": "input", 
    "nameext": ".txt", 
    "size": 1111, 
    "basename": "input.txt", 
    "secondaryFiles": [], 
    "class": "File", 
    "location": "/tmp/tmpSVoLy0/tmpqBzzyJ/tmp3OzwYc/database/files/c/4/2/dataset_c4239d3d-d83f-4c02-8886-43db4da9dbe1.dat"
}
```

> CwlConformanceTestCase.test_conformance_v1_0_nested_workflow
```
E       AssertionError: Request status code (404) was not expected value 200. Body was {u'err_msg': u'Failed to find a tool with uuid [735685f8-c8bc-48c1-95c9-62cd1dacddaa]', u'err_code': 404001}
```

> CwlConformanceTestCase.test_conformance_v1_0_wf_compound_doc
```
E           CompareFail: expected: {
E               "checksum": "sha1$b9214658cc453331b62c2282b772a5c063dbd284", 
E               "class": "File", 
E               "location": "output.txt", 
E               "size": 1111
E           }
E           got: {
E               "basename": "output.txt", 
E               "checksum": "sha1$6552d24faccb0ba3a72c6db11102fc2d399cb627", 
E               "class": "File", 
E               "location": "output.txt", 
E               "nameext": ".txt", 
E               "nameroot": "output", 
E               "size": 1111
E           }
E           caused by: field 'checksum' failed comparison: expected: "sha1$b9214658cc453331b62c2282b772a5c063dbd284"
E           got: "sha1$6552d24faccb0ba3a72c6db11102fc2d399cb627"
```

mr-c committed:
Easy modification of pytest invocation

For example, run only the tests that failed last time:

PYTEST_ADDOPTS=--last-failed tox -e cwl_green

mr-c commented:
Thanks!
